ogr/ogrsf_frmts/miramon/ogrmiramondriver.cpp

```diff
@@ -0,0 +1,249 @@
+                       CPL_UNUSED GDALDataType /*eDT*/, char **papszOptions)
+{
+    OGRMiraMonDataSource *poDS = new OGRMiraMonDataSource();
+
+    if (poDS->Create(pszName, papszOptions))
+        return poDS;
+
+    delete poDS;
+    return nullptr;
+}
+
+/****************************************************************************/
+/*                           RegisterOGRMM()                                */
+/****************************************************************************/
+
+void RegisterOGRMiraMon()
+
+{
+    if (GDALGetDriverByName("MiraMonVector") != nullptr)
+        return;
+
+    GDALDriver *poDriver = new GDALDriver();
+    poDriver->SetDescription("MiraMonVector");
+    poDriver->SetMetadataItem(GDAL_DCAP_VECTOR, "YES");
+    poDriver->SetMetadataItem(GDAL_DCAP_CREATE_LAYER, "YES");
+    poDriver->SetMetadataItem(GDAL_DCAP_CREATE_FIELD, "YES");
+    poDriver->SetMetadataItem(GDAL_DMD_LONGNAME,
+                              "MiraMon Vectors (.pol, .arc, .pnt)");
+    poDriver->SetMetadataItem(GDAL_DMD_EXTENSION, "pol");
+    poDriver->SetMetadataItem(GDAL_DMD_EXTENSIONS, "pol arc pnt");
+    poDriver->SetMetadataItem(GDAL_DMD_HELPTOPIC,
+                              "drivers/vector/miramon.html");
+    poDriver->SetMetadataItem(GDAL_DCAP_VIRTUALIO, "YES");
+    poDriver->SetMetadataItem(GDAL_DCAP_Z_GEOMETRIES, "YES");
+
+    poDriver->SetMetadataItem(
+        GDAL_DMD_OPENOPTIONLIST,
+        "<OpenOptionList>"
+        "  <Option name='Height' scope='vector' type='string' "
+        "   description='Sets which of the possible heights is chosen: "
+        "   the first, the highest or the lowest one.'>"
+        "    <Value>First</Value>"
+        "    <Value>Lower</Value>"
+        "    <Value>Highest</Value>"
+        "  </Option>"
+        "  <Option name='iMultiRecord' scope='vector' type='string' "
+        "   description='Sets which of the possible records is chosen: "
+        "   0, 1, 2,... or the Last one. Use JSON when a serialized "
+        "   JSON is wanted'>"
+        "    <Value>0</Value>"
+        "    <Value>1,...</Value>"
+        "    <Value>Last</Value>"
+        "    <Value>JSON</Value>"
+        "  </Option>"
+        "  <Option name='OpenMemoryRatio' scope='vector' type='float' "
+        "   description='Ratio used to enhance certain aspects of memory"
+        "   In some memory allocations, a block of 256 bytes is used."
+        "   This parameter can be adjusted to achieve"
+        "   OpenMemoryRatio*256."
+        "   For example, OpenMemoryRatio=2 in powerful computers and"
+        "   OpenMemoryRatio=0.5 in less powerful computers."
+        "   By increasing this parameter, more memory will be required,"
+        "   but there will be fewer read/write operations to the disk.'>"
+        "    <Value>0.5</Value>"
+        "    <Value>1</Value>"
+        "    <Value>2</Value>"
+        "  </Option>"
+        "  <Option name='OpenLanguage' scope='vector' type='string' "
```

Comment:
if only the 3 values of the below enumeration are valid, this should be:
```suggestion
        "  <Option name='OpenLanguage' scope='vector' type='string-select' "
```